---
title: "Pythium Experiments data analysis 2024"
author: "Fiama Guevara"
date: "2024-05-21"
output:
  pdf_document: default
  html_document:
    df_print: paged
code_download: yes
editor_options: 
  chunk_output_type: inline
---

```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Setting libraries

library("phyloseq")
library("ggplot2")
library("scales")
library("grid")
library("DESeq2")
library("vegan")
library("ape")
library("plyr")
library("ggpmisc")
library("dplyr")
library("broom")
library("picante")
library("Rmisc")
library("emmeans")
library("car")
library("lme4")
library("tibble")
library("data.table")
#library("limma")
library("microbiome")
library("ggvenn")
library("patchwork")
library("RColorBrewer")
library("ampvis2")
library("microViz")
library("tidyverse")
library("tidyr")
library("parallel")
library("tidyr")
library("stringr")
library("stringi")
library("MuMIn")
library("ggpubr")
library("randomcoloR")
#library("miaViz")
library("metagenomeSeq")
library("cowplot")
library("decontam")
library("microeco")
library("ggthemes")
library(multcomp) #poshoc lmer
library(RVAideMemoire) #permanova pairwise comparison
library(ggnetwork)
library(dada2)
```

# Import dataset
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Importing phyloseq objects
#ps.16S.path <- readRDS('ps.16SPathogens.rds')  #Has 131 samples
sample_data(ps.16SPathogens)

ps.16S.path <- ps.16SPathogens

#ps.ITS.path <- readRDS('ps.ITSPathogens.rds')  #Has 130 samples

## Rename ASVs for 16S (in taxa table and sequences file:
#taxa_names(ps.16S.path) <- paste("ASV", 1:ntaxa(ps.16S.path), sep = "_")
#taxa_names(ps.16S.path)[1:3]

## save the final phyloseq object
#saveRDS(ps.16S.path, 'ps.16SPathogens.rds')

### Export fasta file with sequences and new ASV names
#source("taxa_summary.R", local = TRUE)
#ps_to_fasta(ps.16S.original, out.file = "Bacteria_ASVs.fasta", width = 1000)


# Rename ASVs for ITS (in taxa table and sequences file:
#taxa_names(ps.ITS.path) <- paste("ASV", 1:ntaxa(ps.ITS.path), sep = "_")
#taxa_names(ps.ITS.path)[1:3]

# save the final phyloseq object
#saveRDS(ps.ITS.path, 'ps.ITSPathogens.rds')

### Export fasta file with sequences and new ASV names
#source("taxa_summary.R", local = TRUE)
#ps_to_fasta(ps.ITS.original, out.file = "Fungi_ASVs.fasta", width = 1000)

#sample_data(ps.16S.path)
#sample_data(ps.ITS.path)

```

# Analysis 16S
## Decontamination 16S
```{r, warning=FALSE, message=FALSE, echo=FALSE}
sample_data(ps.16S.path)$is.neg <- sample_data(ps.16S.path)$severity == "ControlN"

contamdf.prev <- isContaminant(ps.16S.path, method = "prevalence", neg = "is.neg", threshold = 0.5)

table(contamdf.prev$contaminant)
#FALSE=7153 #TRUE=79

# Check which taxa are contaminants:
which(contamdf.prev$contaminant)
  #11   28  105  162  302  414  452  467  506  529  569  625  678  797  958 1140 1166 1325 1390 1462 1539 1550 1561 1589 1774 2126 2246 2400 2423 2581 2755 2795 2814 2836 2868 2985 2989 3007 3401 3602 3687 3698 3702 3787 3831 3832 3899 4054 4286 4479 4500 4695 4758 4766 4896 4909 4934 5062 5103 5104 5195 5200 5230 5276 5338 5413 5416 5514 5519 5710 5799 5998 6469 6495 6528 6575 6596 6914 7120

## Contaminants taxa
ps_contamB <- prune_taxa(contamdf.prev$contaminant, ps.16S.path)
tax_table(ps_contamB)

## Remove the identified contaminants 
ps.16S.pathNC <- prune_taxa(!contamdf.prev$contaminant, ps.16S.path)
sample_data(ps.16S.pathNC)

## save the final phyloseq object
#saveRDS(ps.16S.pathNC, 'ps.16SPathogensNC.rds')

```

```{r}
#Poportion of our data were removed
pre.16S <- sum(sample_sums(ps.16S.path))
post.16S <- sum(sample_sums(ps.16S.pathNC))

(pre.16S-post.16S) / pre.16S    #Result=0.1585875
```

## Diagnostic number reads (after QC and filtering ampliseq)
```{r}
# Get out controls
ps.16S.Pythium <- ps.16S.path %>% ps_filter(experiment != "PM", severity != "seedling", run != "1") 
sample_data(ps.16S.Pythium)


#Get reads count
sdt.16S = data.table(as(sample_data(ps.16S.Pythium), "data.frame"),
                 TotalReads = sample_sums(ps.16S.Pythium), keep.rownames = TRUE)
setnames(sdt.16S, "rn", "SampleID")
sdt.16S$TotalReads <- as.numeric(sdt.16S$TotalReads)

sdt.16S %>% 
  arrange(TotalReads) 

```


```{r}
TotalReads_experiment <- sdt.16S %>% 
                        group_by(experiment) %>%
                            summarize(Total = sum(TotalReads))
```

```{r}
MeanReads_experiment <- sdt.16S %>% 
                        group_by(experiment) %>%
                            summarize(Mean = mean(TotalReads))
```

## Sub-setting
```{r}
# Load phyloseq object
#ps.16S.pathNC <- readRDS('ps.16SPathogensNC.rds')

# Add info to metadata (Already saved)
#mappingB <- read.table('16Smetadata_Pathogens.txt', sep = "\t", header = T, row.names = 1)
#sample_data(ps.16S.pathNC) <- mappingB

#saveRDS(ps.16S.pathNC, 'ps.16SPathogensNC.rds')

#Before decontamination has 3233 taxa and 54 samples (72 ASVs difference from 79 identified in all pathogens dataset)
ps.16S.PythiumOrig <- ps.16S.path %>% ps_filter(experiment == "Pythium", severity != "seedling", run != "1")
sample_data(ps.16S.PythiumOrig)


# Get ASVs count after decontamination
ps.16S.Pythium <- ps.16S.pathNC %>% ps_filter(experiment == "Pythium", severity != "seedling", run != "1")
sample_data(ps.16S.Pythium)

#Pythium ps has 3161 taxa and 54 samples


#Remove Archaea Kingdom and Unclassified ASVs
ps.16S.Pythium %>% subset_taxa(Kingdom != "Archaea")

#Modify metadata to create new factor "Inoculation"
new_metadata <- read.table('metadata_Pythium.txt', sep = "\t", header = T, row.names = 1)
sample_data(ps.16S.Pythium) <- new_metadata

#saveRDS(ps.16S.Pythium, 'ps.16S.PythiumNC.rds')

#Load final version phyloseq object
ps.16S.Pythium <- readRDS('ps.16S.PythiumNC.rds') 

#Get number of ASVs assigned to Genus level
taxa <- as.data.frame(tax_table(ps.16S.Pythium))
df2 <- taxa %>%
  mutate_all(as.factor) 

summary(df2)

```

```{r}

##REVIEW!!! NOT UPDATED!!!!!

### Save and Export fasta file with sequences from Phylum=NAs
Pythium_NAs <- ps.16S.Pythium %>% subset_taxa(Phylum == "")

taxa.16S.Pythium <- tax_table(ps.16S.Pythium)
#write.table(taxa.16S.Pythium, file="taxa.16S.Pythium.txt") 

taxa.16S.Pythium <- as.data.frame(taxa.16S.Pythium)
#View(taxa.PythiumNAs)
#class(taxa.PythiumNAs)

seqs.16S.Py <- Biostrings::DNAStringSet(as.vector(taxa.16S.Pythium$sequence)) 
names(seqs.16S.Py) <- row.names(taxa.16S.Pythium) 

Biostrings::writeXStringSet(seqs.16S.Py, filepath = "Pythium.16S.ASVs.fasta", width = 1000)

```
## Rarefaction curves
```{r, warning=FALSE, message=FALSE, echo=FALSE}

theme_set(theme_bw(base_size = 14))

metadata <- data.frame(sample_data(ps.16S.Pythium), check.names = FALSE) |>
  rownames_to_column("sample_id")
asv_table <- data.frame(otu_table(ps.16S.Pythium), tax_table(ps.16S.Pythium),
                        check.names = FALSE)
ps_amp <- amp_load(taxonomy = tax_table(ps.16S.Pythium),
                   otutable = otu_table(ps.16S.Pythium),
                   metadata = metadata)
                    
Bact_curves <- amp_rarecurve(ps_amp, stepsize = 1000,
              color_by = "severity",
              facet_by = "run") +
  labs(x = "\nTotal ASV count (library size)",
       y = "\nNumber of distinct ASVs (richness)",
       color = "severity") +
  scale_x_continuous(expand = c(0, 0), labels = scales::comma) +
  scale_y_continuous(expand = c(0, 0))

# Save plot
ggsave("Bacteria_Rarefcurves.png", Bact_curves, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)

```

## Taxonomic composition Controls
```{r, warning=FALSE, message=FALSE, echo=FALSE}

# Agglomerate taxa by phylum level
ps.16S.Pythium.P <- tax_glom(ps.16S.Pythium, taxrank = 'Phylum', NArm = FALSE)

# get Relative abundance
ps.16S.Pythium_RA <- transform_sample_counts(ps.16S.Pythium.P, function(x) x/sum(x))

# Create dataframe from phyloseq object
data_ps.16S.Pythium <- psmelt(ps.16S.Pythium_RA)

#Rename less abundant - simple way to rename taxa with < 5% abundance
data_ps.16S.Pythium$Phylum[data_ps.16S.Pythium$Abundance < 0.05] <- "Phylum < 5% abund."

#Bar plot by Phylum level
# Set color palette to accommodate the number of phylum
colourCount = length(unique(data_ps.16S.Pythium$Phylum))
set.seed(2020)
taxaColors <- distinctColorPalette(colourCount)

# Plot
plot.phylum <- ggplot(data=data_ps.16S.Pythium, aes(x=severity, y=Abundance, fill = fct_reorder(Phylum, Abundance))) + theme(legend.position="right")

final.plot.phylum <-plot.phylum +
  geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values=taxaColors) +
  labs(x = "Disease severity", y = "Relative Abundance\n") +
  guides(fill=guide_legend(title="Phylum")) 
  
#ggsave("Rplots_survey/taxa.phylum.16S.png", final.plot.phylum, 
  #width = 20,
  #height = 20,
  #units = "cm",
  #dpi = 600)

```

## CSS normalization
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Convert the phyloseq object to a metagenomeseq object:
mgs_css <- phyloseq_to_metagenomeSeq(ps.16S.Pythium)

# Perform the Cumulative Sum Scaling:
mgs_css <- cumNorm(mgs_css)

# Extract the counts and add them to a separate phyloseq object:
css_counts <- MRcounts(mgs_css, norm = TRUE)
ps_css <- ps.16S.Pythium
otu_table(ps_css) <- otu_table(t(css_counts), taxa_are_rows = FALSE)

#Now lets compare the original data to the CSS normalized data:
otu_table(ps.16S.Pythium)[1:5, 1:5]
head(sample_sums(ps.16S.Pythium))

otu_table(ps_css)[1:5, 1:5]
head(sample_sums(ps_css))

```

## Taxonomic composition 16S  NEED TO REVISE!!!
```{r, warning=FALSE, message=FALSE, echo=FALSE}

# Agglomerate taxa by phylum level
ps.16S.Pythium.P <- tax_glom(ps.16S.Pythium, taxrank = 'Phylum', NArm = FALSE)

# get Relative abundance
ps.16S.Pythium_RA <- transform_sample_counts(ps.16S.Pythium.P, function(x) x/sum(x))

# Create dataframe from phyloseq object
data_ps.16S.Pythium <- psmelt(ps.16S.Pythium_RA)

#Rename less abundant - simple way to rename taxa with < 5% abundance
data_ps.16S.Pythium$Phylum[data_ps.16S.Pythium$Abundance < 0.05] <- "Phylum < 5% abund."

#Bar plot by Phylum level
# Set color palette to accommodate the number of phylum
colourCount = length(unique(data_ps.16S.Pythium$Phylum))
set.seed(2020)
taxaColors <- distinctColorPalette(colourCount)

# Plot
plot.phylum <- ggplot(data=data_ps.16S.Pythium, aes(x=severity, y=Abundance, fill = fct_reorder(Phylum, Abundance))) + theme(legend.position="right")

final.plot.phylum <-plot.phylum +
  geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values=taxaColors) +
  labs(x = "Disease severity", y = "Relative Abundance\n") +
  guides(fill=guide_legend(title="Phylum")) 
  
#ggsave("Rplots_survey/taxa.phylum.16S.png", final.plot.phylum, 
  #width = 20,
  #height = 20,
  #units = "cm",
  #dpi = 600)

```

## Alpha diversity
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#Load phyloseq object
ps.16S.Pythium <- readRDS('ps.16S.PythiumNC.rds')

# Alpha diversity - Evenness
Evenness.16S.Pythium <- microbiome::evenness(ps.16S.Pythium, index = "all")

# Create data frame with evenness measurements
df.16S.Pythium.Ev <- Evenness.16S.Pythium[,c('pielou', 'simpson')]

# All alpha diversity indexes estimation
richness.16S.Pythium <- estimate_richness(ps.16S.Pythium, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
                                  
#Create a data frame with All alpha diversity and metadata
adiv.16S.Pythium <- cbind(sample_data(ps.16S.Pythium), richness.16S.Pythium, df.16S.Pythium.Ev)

#write.table(adiv.16S.Pythium, file="adiv.all.16S.PythiumResults.txt") 

#adiv.16S.Pythium <- read.table("adiv.all.16S.PythiumResults.txt")
```

```{r}
### Means and standard deviation diversity indexes by disease severity
mean_sd_adiv.16S <- adiv.16S.Pythium %>%
  group_by(severity, run) %>%
  summarise_at(vars(c(Observed, Shannon, InvSimpson, Chao1, pielou)), list(name = mean, sd))

#write.table(mean_sd_adiv.16S, file="mean_sd_adiv.16SPythium.txt")
```


```{r}
################### Statistics - alpha diversity ################################
### Diagnostic models

####### Observed (richness)
model.Observed <- lmer(Observed ~ severity + (1|run), data = adiv.16S.Pythium)
Anova(model.Observed) #<0.001 significant effect of severity and run and interaction

###Analysis by run
#Run 1
adiv.16S.PythiumR1 <- adiv.16S.Pythium %>% filter(run == "2")

model.ObservedR1 <- lmer(Observed ~ severity + (1|pond), data = adiv.16S.PythiumR1)
Anova(model.ObservedR1) #<significant effect of severity p-val=2.264e-12 ***

#Residuals plots
par(mfrow = c(2, 3))
plot(model.ObservedR1, which=1:6) 
res <- resid(model.ObservedR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal  p-value = 0.1475

# Pairwise comparisons
mObs <- emmeans(model.ObservedR1, "severity")
pairs(mObs) #significant differences between control-high and high-low

multcomp::cld(object = mObs, Letters = letters)


#####Run 2
adiv.16S.PythiumR2 <- adiv.16S.Pythium %>% filter(run == "3")

model.ObservedR2 <- lmer(Observed ~ severity + (1|pond), data = adiv.16S.PythiumR2)
Anova(model.ObservedR2) #NO significant effect of severity p-val=0.3672

#Residuals plots
par(mfrow = c(2, 3))
plot(model.ObservedR2, which=1:6) 
res <- resid(model.ObservedR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal  p-value = 0.3152

# Pairwise comparisons
mObs <- emmeans(model.ObservedR2, "severity")
pairs(mObs) #NO significant differences between any

multcomp::cld(object = mObs, Letters = letters)


# poshoc
compare_means(Observed ~ severity, data = adiv.16S.PythiumR2, method = "t.test")  
#sig difference any

```


```{r}
######### Simpson index
model.Simpson <- lm(InvSimpson ~ severity * run, data = adiv.16S.Pythium) 
Anova(model.Simpson)  # significant effect of severity*run (p-val=<0.0001)

###Run1 
model.SimpsonR1 <- lmer(InvSimpson ~ severity + (1|pond), data = adiv.16S.PythiumR1) 
Anova(model.SimpsonR1) # significant effect of severity p-val=0.002052 **

#Residuals plots
plot(model.SimpsonR1, which=1:6) 
res <- resid(model.SimpsonR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.7485

# Pairwise comparisons
mSS <- emmeans(model.SimpsonR1, "severity")
pairs(mSS) #Almost sig diff control-high and control-low

multcomp::cld(object = mSS, Letters = letters)


###Run 2
model.SimpsonR2 <- lmer(InvSimpson ~ severity + (1|pond), data = adiv.16S.PythiumR2) 
Anova(model.SimpsonR2) # NO significant effect of severity p-val=0.129

#Residuals plots
plot(model.SimpsonR2, which=1:6) 
res <- resid(model.SimpsonR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.3275

# Pairwise comparisons
mSS <- emmeans(model.SimpsonR2, "severity")
pairs(mSS) #no sig diff bc interaction

multcomp::cld(object = mSS, Letters = letters)



```


```{r}
####### Shannon index
model.Shannon <- lm(Shannon ~ severity * run, data = adiv.16S.Pythium)
Anova(model.Shannon) #significant effect of severity*run (p-val=0.0000007853 ***)


##Run 1
model.ShannonR1 <- lmer(Shannon ~ severity + (1|pond), data = adiv.16S.PythiumR1)
Anova(model.ShannonR1) #significant effect of severity p-val=0.01969 *


#Residuals plots
plot(model.ShannonR1, which=1:6) 
res <- resid(model.ShannonR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.1225


# Pairwise comparisons
mShaS <- emmeans(model.ShannonR1, "severity")
pairs(mShaS) #No Sig dif ??

emmeans(model.ShannonR1, list(pairwise ~ severity))

# poshoc
compare_means(Shannon ~ severity, data = adiv.16S.PythiumR1, method = "anova")  
#sig difference p-val=<0.0001

fm1 <- aov(Shannon ~ severity, data = adiv.16S.PythiumR1)
summary(fm1)
thsd <- TukeyHSD(fm1, "severity", ordered = TRUE)
tidy(thsd)
#sig difference high-control


##Run 2
model.ShannonR2 <- lmer(Shannon ~ severity + (1|pond), data = adiv.16S.PythiumR2)
Anova(model.ShannonR2) #NO significant effect of severity p-val=0.3621


#Residuals plots
plot(model.ShannonR2, which=1:6) 
res <- resid(model.ShannonR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.2401


# Pairwise comparisons
mShaS <- emmeans(model.ShannonR2, "severity")
pairs(mShaS) #No Sig dif

emmeans(model.ShannonR2, list(pairwise ~ severity))


# poshoc
compare_means(Shannon ~ severity, data = adiv.16S.PythiumR2, method = "anova")  
#NO sig difference p-val=0.036

fm1 <- aov(Shannon ~ severity, data = adiv.16S.PythiumR2)
summary(fm1)
thsd <- TukeyHSD(fm1, "severity", ordered = TRUE)
tidy(thsd)
#sig difference high-control

```


```{r}
##### Evenness (Pielou index)
model.Ev <- lm(pielou ~ severity * run, data = adiv.16S.Pythium)
Anova(model.Ev) # significant effect of severity*run (p-val=0.0001369 ***)

##Run 1
model.EvR1 <- lmer(pielou ~ severity + (1|pond), data = adiv.16S.PythiumR1)
Anova(model.EvR1) # significant effect of severity (p-value=0.03563 *)

#Residuals plots
plot(model.EvR1, which=1:6) 
res <- resid(model.EvR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)  ###A Bimodal distribution is observed correlates with each run (potential differences only in one run)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.3014

# Pairwise comparisons
mES <- emmeans(model.EvR1, "severity")
pairs(mES)  #No Sig dif because interactions
multcomp::cld(object = mES, Letters = letters)

# poshoc
compare_means(pielou ~ severity, data = adiv.16S.PythiumR1, method = "t.test")  
#sig diff between all

##Run 2
model.EvR2 <- lmer(pielou ~ severity + (1|pond), data = adiv.16S.PythiumR2)
Anova(model.EvR2) # NO significant effect of severity (p-value=0.3126)

#Residuals plots
plot(model.EvR2, which=1:6) 
res <- resid(model.EvR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)  ###A Bimodal distribution is observed correlates with each run (potential differences only in one run)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.8637

# Pairwise comparisons
mES <- emmeans(model.EvR2, "severity")
pairs(mES)  #No Sig dif because interactions
multcomp::cld(object = mES, Letters = letters)

# poshoc
compare_means(pielou ~ severity, data = adiv.16S.PythiumR2, method = "t.test")  
#sig diff between all

```
#### Boxplot
```{r}
#Boxplot observed alpha diversity 
adiv.16S.Pythium$severity <- ordered(adiv.16S.Pythium$severity, levels = c("low", "control", "high"))

# Prepare a vector of colors with specific color for sample type
library(RColorBrewer)
myColors <- c("palegreen1", "#1F78B4", "hotpink1")

#Observed plot 
observed.16S.Pythium.plot <- ggplot(adiv.16S.Pythium, aes(x = severity, y = Observed, fill=severity)) +
  geom_boxplot() + 
  ylab('Bacteria Richness (Observed)') +
  xlab('Disease Severity') +
  scale_fill_manual(name = "Severity", values = myColors) +
  labs(fill = 'severity') +
  theme(axis.text.x = element_text(angle = 45, hjust=1))

#Plot Shannon
Shannon.16S.Pythium.plot <- ggplot(adiv.16S.Pythium, aes(x = severity, y = Shannon, fill=severity)) + geom_boxplot(outlier.shape = NA) +
  ylab('Shannon index') +
  xlab('Disease Severity') +
  scale_fill_manual(name = "severity", values = myColors) + 
    theme_few() +
    theme(legend.position = "none")

# Save plot
ggsave("Pythium_ShannonBacteria.png", Shannon.16S.Pythium.plot, 
  width = 11,
  height = 12,
  units = "cm",
  dpi = 600)

#Plots Evenness 
PielouEv.16S.Pythium.plot <- ggplot(adiv.16S.Pythium, aes(x = severity, y = pielou, fill=severity)) +
  geom_boxplot() + 
  ylab('Bacteria Evenness (Pielou)') +
  xlab('Disease Severity') +
  scale_fill_manual(name = "severity", values = myColors) +
  labs(fill = 'severity') 

# Merging plots
library(patchwork)

alphaDiv16S.Pythium <- observed.16S.Pythium.plot + PielouEv.16S.Pythium.plot +
  #plot_annotation(tag_levels = "A") +
    plot_layout(guides = 'collect') & 
    theme (axis.title = element_text (size = 10)) &
      theme(plot.tag = element_text(size = 12)) &
          theme(axis.text.y = element_text (size = 8)) &
            scale_x_discrete(name ="Disease severity") 


```

## Beta Diversity
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#Get relative abundance after normalization
#ps_css_RA <- transform_sample_counts(ps_css, function(x) x/sum(x))

ps.16S.Pythium_RA <- transform_sample_counts(ps.16S.Pythium, function(x) x/sum(x))

#Distance
sampledf <- data.frame(sample_data(ps.16S.Pythium_RA))
dist.mat <- phyloseq::distance(ps.16S.Pythium_RA, method = "bray")

set.seed(100)
perm <- how(nperm = 999)
setBlocks(perm) <- with(sampledf, run)

pmv.Py <- adonis2(dist.mat ~ severity, data = sampledf, permutations = perm, by = "margin")
print(pmv.Py)


## Permanova poshoc
set.seed(100)
pairwise.perm.manova(dist.mat, sampledf$severity, nperm = 999, progress = TRUE, p.method = "fdr", F = T, R2 = T) #Sig diff control-high and control-low p-value = 0.003 and high-low p-value=0.992


pairwise.perm.manova(dist.mat, sampledf$run, nperm = 999, progress = TRUE, p.method = "fdr", F = T, R2 = T) #p-val=0.001

pairwise.perm.manova(dist.mat, sampledf$inoculation, nperm = 999, progress = TRUE, p.method = "fdr", F = T, R2 = T) #p-val=0.001

```

### PCoA plot
```{r}
#Color palette
myColors2 <- c("#1F78B4", "hotpink1", "palegreen1")

#Calculate ordinations
ordBC.Py = ordinate(ps_css_RA, method = "PCoA", distance = dist.mat.Py)

## Plot
p.Ord.BC.Py <- plot_ordination(ps_css_RA, ordBC.Py, color = "severity") 

Bdiv.plo.Py <- p.Ord.BC.Py + 
  geom_point(mapping = aes(shape = factor(run)),size=5, alpha=0.85) +
    stat_ellipse() +
      geom_vline(xintercept = c(0), color = "grey70", linetype = 2) +
        geom_hline(yintercept = c(0), color = "grey70", linetype = 2) +
          theme_bw() + 
            theme(panel.background = element_blank(), 
              panel.grid = element_blank(), 
              legend.title=element_text(size=14),
              legend.text=element_text(size=12)) + 
  scale_color_manual(name = "Disease severity", values = myColors2, breaks =c("control", "high", "low"), labels = c("Control", "High", "Low")) +
  scale_shape_discrete(labels=c("Run 1","Run 2"),name="Experiment")
                                    
# Save plot
ggsave("PCoA_Bray_Pyhtium.png", Bdiv.plo.Py, 
  width = 15,
  height = 12,
  units = "cm",
  dpi = 600)


```

## Beta-dispersion test
```{r}
# calculates the beta-dispersion for each group, when comparing 2 or more
#For severity
disp.testSev <- betadisper(dist.mat, group = sampledf$severity, bias.adjust = TRUE)

#For inoculation
disp.testInc <- betadisper(dist.mat, group = sampledf$inoculation, bias.adjust = TRUE)

#For run
disp.testRun <- betadisper(dist.mat, group = sampledf$run, bias.adjust = TRUE)

# tests if centroid distances are significantly different from each other
disp.anovaSev <- anova(disp.testSev) #Result= centroids NO sig different p-value=0.1171

disp.anovaInc <- anova(disp.testInc) #Result= centroids sig different p-value=0.01798 *

disp.anovaRun <- anova(disp.testRun) #Result= centroids NO sig different p-value=0.3591
```

```{r}
# test significance between each group
dispSev.TukeyHSD <- TukeyHSD(disp.testSev) #sig diff high-control (p-adj=0.0144541), low-control (p-adj=0.0175967) but NO for high-low (p-adj=0.9969173)
dispInc.TukeyHSD <- TukeyHSD(disp.testInc) #sig diff p-adj=0.0006948

dispRun.TukeyHSD <- TukeyHSD(disp.testRun) #NO sig diff p-adj=0.972428

```

```{r}
# plot showing the dispersion for each group
plot(disp.testInc, hull = FALSE, ellipse = TRUE)
```

```{r}
#test for homogeneity of multivariate dispersions 
set.seed(100)
permutest(disp.testInc, pairwise = TRUE, permutations = 999) #p-val=0.016 *
```

## Differential abundance analysis
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#REFERENCE FOR ADDITIONAL PLOTS: https://rpubs.com/mrgambero/1025383

# Select only taxa present on 75% data at abundance >1% 
#Remove rare taxa Remove taxa not seen more than 3 times in at least 20% of the samples. This protects against an OTU with small mean & trivially large C.V.

#physeq1ft <- filter_taxa(physeq1, function(x) sum(x > 3) > (0.2*length(x)), TRUE)

#filter_taxa(ps.its, function (x) {sum(x > 0) > 1}, prune=TRUE)

# Agglomerate taxa
ps.16S.Pythium <- tax_glom(ps.16S.Pythium, "Genus")

#Fix taxa names
#we want to add the genus, and if not available, the last taxonomic information available.
ps.16S.Pythium <- tax_fix(ps.16S.Pythium)

# Converting phyloseq object to DESeq2 format 
dds.16S.Pythium.df <- phyloseq_to_deseq2(ps.16S.Pythium, ~severity + run)

# Identifying the groups in our experiment
dds.16S.Pythium.df$Group <- factor(paste0(dds.16S.Pythium.df$severity, dds.16S.Pythium.df$run))  

# Setting our experimental design in the DESeq2 object
design(dds.16S.Pythium.df) <- formula(~Group) 

# Generating our analysis
dds.analysis <- DESeq(dds.16S.Pythium.df, quiet = TRUE) 

results(dds.analysis)

# To check contrasting groups
levels(dds.analysis$Group)

### Select the groups to contrast 
# Contrast Control vs. High in Run 2
res.CvsH.run2 <- results(dds.analysis, contrast = c("Group", "control2", "high2")) 
# Contrast Control vs. Low in Run 2
res.CvsL.run2 <- results(dds.analysis, contrast = c("Group", "control2", "low2")) 

# Contrast Control vs. High in Run 3
res.CvsH.run3 <- results(dds.analysis, contrast = c("Group", "control3", "high3")) 
# Contrast Control vs. Low in Run 3
res.CvsL.run3 <- results(dds.analysis, contrast = c("Group", "control3", "low3"))

```

```{r}
## RESULTS CONTRAST: Contrast Control vs. High in Run 2
# First look at the results 
summary(res.CvsH.run2) 

# Number of taxa with significantly higher abundance in High run2:
sum(res.CvsH.run2$padj < 0.1 & res.CvsH.run2$log2FoldChange > 0, na.rm = TRUE)  #Result 48

# Number of taxa with significantly lower abundance in High run2:
sum(res.CvsH.run2$padj < 0.1 & res.CvsH.run2$log2FoldChange < 0, na.rm = TRUE)  #Result 35

# To look the most significantly different taxa
res.CvsH.run2.order <- res.CvsH.run2[order(res.CvsH.run2$pvalue), ]

# Create a data frame from your DESeq results 
res.CvsH.run2.df <- as.data.frame(res.CvsH.run2)

### Add the taxonomy of the ASVs to the new data frame 
CvsH.run2.taxmelt <- as.data.frame(tax_table(ps.16S.Pythium))
## now let us apply a the prefix to each one of them
res.CvsH.run2.df$Tax <- CvsH.run2.taxmelt$Genus
  
  #paste(rownames(res.CvsH.run2.df), CvsH.run2.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsH.run2.df$association = ifelse(res.CvsH.run2.df$padj <= 0.05 & res.CvsH.run2.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsH.run2.df$association = ifelse(res.CvsH.run2.df$padj <= 0.05 & res.CvsH.run2.df$log2FoldChange < 0, "Low_abundant", res.CvsH.run2.df$association)

# Filter to keep only log2folds > |5| and pvalues <= 0.05 
CvsH.run2.res05 <- dplyr::filter(res.CvsH.run2.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Making sure our new table has everything we need 
head(CvsH.run2.res05)

# Generating a dataframe from our table
CvsH.run2.res05.df <- as.data.frame(CvsH.run2.res05)

#Add new variables for categories used while plotting
CvsH.run2.res05.df$Severity <- "High"
CvsH.run2.res05.df$Exp_run <- "Run2"

## Bar plot 
CvsH.run2.plot <- ggplot(data = CvsH.run2.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Bacterial Genera in High Disease Severity vs. Control - Run2") +
  theme(plot.title = element_text(hjust = 0.5))

```

```{r}
## RESULTS CONTRAST: Contrast Control vs. Low in Run 2

# First look at the results 
summary(res.CvsL.run2)

# Number of taxa with significantly higher abundance in Low run2:
sum(res.CvsL.run2$padj < 0.1 & res.CvsL.run2$log2FoldChange > 0, na.rm = TRUE)  #Result 49

# Number of taxa with significantly lower abundance in Low run2:
sum(res.CvsL.run2$padj < 0.1 & res.CvsL.run2$log2FoldChange < 0, na.rm = TRUE)  #Result 34

# To look the most significantly different taxa
res.CvsL.run2.order <- res.CvsL.run2[order(res.CvsL.run2$pvalue), ]
head(res.CvsL.run2.order)

# Create a data frame from your DESeq results 
res.CvsL.run2.df <- as.data.frame(res.CvsL.run2)

# Add the taxonomy of the ASVs to the new data frame 
CvsL.run2.taxmelt <- as.data.frame(tax_table(ps.16S.Pythium))
res.CvsL.run2.df$Tax <- CvsL.run2.taxmelt$Genus
  
  #paste(rownames(res.CvsL.run2.df), CvsL.run2.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsL.run2.df$association = ifelse(res.CvsL.run2.df$padj <= 0.05 & res.CvsL.run2.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsL.run2.df$association = ifelse(res.CvsL.run2.df$padj <= 0.05 & res.CvsL.run2.df$log2FoldChange < 0, "Low_abundant", res.CvsL.run2.df$association)


# Filter to keep only log2folds > |5| and pvalues <= 0.05 
CvsL.run2.res05 <- dplyr::filter(res.CvsL.run2.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Making sure our new table has everything we need 
head(CvsL.run2.res05)

# Generating a dataframe from our table
CvsL.run2.res05.df <- as.data.frame(CvsL.run2.res05)

#Add new variables for categories used while plotting
CvsL.run2.res05.df$Severity <- "Low"
CvsL.run2.res05.df$Exp_run <- "Run2"

## Plot 
CvsL.run2.plot <- ggplot(data = CvsL.run2.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Bacterial Genera in Low Disease Severity vs. Control - Run2") +
  theme(plot.title = element_text(hjust = 0.5))

```

### Barplot High&Low Run2
```{r}
# Join dataframes with DeSeq results
BindDF.run2 <- bind_rows(CvsH.run2.res05.df, CvsL.run2.res05.df)

##Save final dataframe
write.table(BindDF.run2, file = "DESeq_Pythium_data.tsv", sep = "\t", row.names = TRUE)

## Plot (normal baplot - don't show all ASVs from low and high disease severity)
finalRun2.plot <- ggplot(data = BindDF.run2,
       aes(x = log2FoldChange, y = Tax, fill = Severity, colour = association)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  geom_vline(xintercept = c(0), color = "grey0", linetype = 2) +
  labs(title = "Experiment Run 1") +
  scale_fill_manual(name = "Disease severity", values = c("white", "grey")) +
  scale_color_manual(name = "Abundance", values = c("tomato1", "steelblue3"), breaks = c("High_abundant", "Low_abundant"), labels = c("Enriched in inoculated roots", "Depleted in inoculated roots")) + labs(y = "\nGenera") +
   theme_few() +
  theme(plot.title = element_text(size = 12),
  theme(axis.text.x = element_text(angle = 90, size = 13),
        legend.title=element_text(size=13),
        legend.text=element_text(size=12))) 
 

# Save plot
ggsave("DiffAbundBacteria-Pythium.png", finalRun2.plot, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)
  

# Use alpha to distinguish between low and high (worked partially)
ggplot() + 
  geom_bar(data=BindDF.run2[which(BindDF.run2$Severity == "High"),], aes(log2FoldChange, Tax, fill = association),stat = 'identity',position="dodge") +
  geom_bar(data=BindDF.run2[which(BindDF.run2$Severity == "Low"),], aes(log2FoldChange, Tax, fill = association),stat = 'identity',position = "dodge",alpha=0.5) + coord_flip() +
  labs(title = "Differentially abundant Bacteria\nExperiment Run 1") 
```

```{r}
## RESULTS CONTRAST: Contrast Control vs. High in Run 3

# First look at the results 
summary(res.CvsH.run3)

# Create a data frame from your DESeq results 
res.CvsH.run3.df <- as.data.frame(res.CvsH.run3)

# Add the taxonomy of the ASVs to the new data frame 
CvsH.run3.taxmelt <- as.data.frame(tax_table(ps.16S.Pythium))
res.CvsH.run3.df$Tax <- CvsH.run3.taxmelt$Genus
  
  #paste(rownames(res.CvsH.run3.df), CvsH.run3.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsH.run3.df$association = ifelse(res.CvsH.run3.df$padj <= 0.05 & res.CvsH.run3.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsH.run3.df$association = ifelse(res.CvsH.run3.df$padj <= 0.05 & res.CvsH.run3.df$log2FoldChange < 0, "Low_abundant", res.CvsH.run3.df$association)


# Filter to keep only log2folds > |5| and pvalues <= 0.05 
CvsH.run3.res05 <- dplyr::filter(res.CvsH.run3.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Generating a dataframe from our table
CvsH.run3.res05.df <- as.data.frame(CvsH.run3.res05)

#Add new variables for categories used while plotting
CvsH.run3.res05.df$Severity <- "High"
CvsH.run3.res05.df$Exp_run <- "Run3"

## Plot 
CvsH.run3.plot <- ggplot(data = CvsH.run3.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Bacterial Genera in High Disease Severity vs. Control - Run3") +
  theme(plot.title = element_text(hjust = 0.5))

```

```{r}
## RESULTS CONTRAST: Contrast Control vs. Low in Run 3

# First look at the results 
summary(res.CvsL.run3)

# Create a data frame from your DESeq results 
res.CvsL.run3.df <- as.data.frame(res.CvsL.run3)

# Add the taxonomy of the ASVs to the new data frame 
CvsL.run3.taxmelt <- as.data.frame(tax_table(ps.16S.Pythium))
res.CvsL.run3.df$Tax <- CvsL.run3.taxmelt$Genus
  
  #paste(rownames(res.CvsL.run3.df), CvsL.run3.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsL.run3.df$association = ifelse(res.CvsL.run3.df$padj <= 0.05 & res.CvsL.run3.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsL.run3.df$association = ifelse(res.CvsL.run3.df$padj <= 0.05 & res.CvsL.run3.df$log2FoldChange < 0, "Low_abundant", res.CvsL.run3.df$association)


# Filter to keep only log2folds > |5| and pvalues <= 0.05 
CvsL.run3.res05 <- dplyr::filter(res.CvsL.run3.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Generating a dataframe from our table
CvsL.run3.res05.df <- as.data.frame(CvsL.run3.res05)

#Add new variables for categories used while plotting
CvsL.run3.res05.df$Severity <- "Low"
CvsL.run3.res05.df$Exp_run <- "Run3"

## Plot 
CvsL.run3.plot <- ggplot(data = CvsL.run3.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Bacterial Genera in Low Disease Severity vs. Control - Run3") +
  theme(plot.title = element_text(hjust = 0.5))

```

### Barplot High&Low Run3
```{r}
# Join dataframes with DeSeq results
BindDF.run3 <- bind_rows(CvsH.run3.res05.df, CvsL.run3.res05.df)

##Save final dataframe
write.table(BindDF.run3, file = "DESeq_Pythium_dataRun3.tsv", sep = "\t", row.names = TRUE)

## Plot (normal baplot - don't show all ASVs from low and high disease severity)
finalRun3.plot <- ggplot(data = BindDF.run3,
       aes(x = log2FoldChange, y = Tax, fill = Severity, colour = association)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  geom_vline(xintercept = c(0), color = "grey0", linetype = 2) +
  labs(title = "Experiment Run 2") +
  scale_fill_manual(name = "Disease severity", values = c("white", "grey")) +
  scale_color_manual(name = "Abundance", values = c("tomato1", "steelblue3"), breaks = c("High_abundant", "Low_abundant"), labels = c("Enriched in inoculated roots", "Depleted in inoculated roots")) + labs(y = "\nGenera") +
   theme_few() +
  theme(plot.title = element_text(size = 12),
  theme(axis.text.x = element_text(angle = 90, size = 13),
        legend.title=element_text(size=13),
        legend.text=element_text(size=12)))
 

# Save plot
ggsave("DiffAbundBacteria-PythiumR3.png", finalRun3.plot, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)
  
```
### All Barplots 
```{r}
# Join dataframes with DeSeq results
BindDF.run23 <- bind_rows(CvsH.run2.res05.df, CvsL.run2.res05.df, CvsH.run3.res05.df, CvsL.run3.res05.df)

# Tidy dataframe
data.run23 <- BindDF.run23 %>%
  dplyr::group_by(Tax, association, Severity, Exp_run) %>% 
  dplyr::summarize(log2FoldChange = mean(log2FoldChange))

data.run23 <- aggregate(log2FoldChange ~ ., BindDF.run23, sum)

##Save final dataframe
write.table(data.run23, file = "DESeq_Pythium_dataRun23.tsv", sep = "\t", row.names = TRUE)

## Plot (normal baplot - don't show all ASVs from low and high disease severity)
finalRun23.plot <- ggplot(data = data.run23,
       aes(x = log2FoldChange, y = Tax, fill = Severity)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  coord_flip() +
  facet_grid(~Exp_run) +
  geom_vline(xintercept = c(0), color = "grey70", linetype = 2) +
  labs(title = "Differentially abundant Bacteria") +
  scale_fill_manual(name = "Disease severity", values = c("tomato1", "lightsalmon")) + labs(y = "\nGenera") + 
  theme(strip.background = element_rect(color="black", fill="gray", size=1, linetype="solid")) +
  theme_bw() +
  theme(plot.title = element_text(size = 12)) +
  theme(axis.text.x = element_text(angle = 90, size = 7))
 

# Save plot
ggsave("DiffAbundBacteria-PythiumAll.png", finalRun23.plot, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)
  
```

# Analysis ITS
## Decontamination ITS
```{r, warning=FALSE, message=FALSE, echo=FALSE}
sample_data(ps.ITS.path)$is.neg <- sample_data(ps.ITS.path)$severity == "ControlN"

contamdfF.prev <- isContaminant(ps.ITS.path, method = "prevalence", neg = "is.neg", threshold = 0.5)

table(contamdfF.prev$contaminant)
#FALSE=1773 #TRUE=30

# Check which taxa are contaminants:
which(contamdfF.prev$contaminant)
  #ASVs: 75  131  171  287  291  307  394  400  432  499  616  670  779  786 819 1007 1039 1097 1207 1316 1336 1373 1543 1579 1663 1673 1731 1781 1790 1795

## Contaminants taxa
ps_contamF <- prune_taxa(contamdfF.prev$contaminant, ps.ITS.path)
tax_table(ps_contamF)

## Remove the identified contaminants 
ps.ITS.pathNC <- prune_taxa(!contamdfF.prev$contaminant, ps.ITS.path)
sample_data(ps.ITS.pathNC)

## save the final phyloseq object
#saveRDS(ps.ITS.pathNC, 'ps.ITSPathogensNC.rds')

```

## Sub-setting
```{r}
# Load phyloseq object
ps.ITS.pathNC <- readRDS('ps.ITSPathogensNC.rds')

# Add info to metadata (Already saved)
#mappingB <- read.table('ITSmetadata_Pathogens.txt', sep = "\t", header = T, row.names = 1)
#sample_data(ps.ITS.pathNC) <- mappingB

#saveRDS(ps.ITS.pathNC, 'ps.ITSPathogensNC.rds')


#Before decontamination has 464 taxa and 54 samples (72 ASVs difference from 79 identified in all pathogens dataset)
ps.ITS.PythiumOrig <- ps.ITS.path %>% ps_filter(experiment == "Pythium", severity != "seedling", run != "1")
sample_data(ps.ITS.PythiumOrig)


# Get ASVs count after decontamination
ps.ITS.Pythium <- ps.ITS.pathNC %>% ps_filter(experiment == "Pythium", severity != "seedling", run != "1")
sample_data(ps.ITS.Pythium)

#Pythium ps has 435 taxa and 54 samples


#Modify metadata to create new factor "Inoculation"
new_metadata <- read.table('metadata_Pythium.txt', sep = "\t", header = T, row.names = 1)
sample_data(ps.ITS.Pythium) <- new_metadata

#saveRDS(ps.ITS.Pythium, 'ps.ITS.PythiumNC.rds')
ps.ITS.Pythium <- readRDS('ps.ITS.PythiumNC.rds')

#Get number of ASVs assigned to Genus level
taxa <- as.data.frame(tax_table(ps.ITS.Pythium))
df2 <- taxa %>%
  mutate_all(as.factor) 

summary(df2)

#Other form
table(tax_table(ps.ITS.Pythium)[, "Genus"]) 

```

## Rarefaction curves
```{r, warning=FALSE, message=FALSE, echo=FALSE}

theme_set(theme_bw(base_size = 14))

metadata <- data.frame(sample_data(ps.ITS.Pythium), check.names = FALSE) |>
  rownames_to_column("sample_id")
asv_table <- data.frame(otu_table(ps.ITS.Pythium), tax_table(ps.ITS.Pythium),
                        check.names = FALSE)
ps_amp <- amp_load(taxonomy = tax_table(ps.ITS.Pythium),
                   otutable = otu_table(ps.ITS.Pythium),
                   metadata = metadata)
                    
Fungi_curves <- amp_rarecurve(ps_amp, stepsize = 1000,
              color_by = "severity",
              facet_by = "run") +
  labs(x = "\nTotal ASV count (library size)",
       y = "\nNumber of distinct ASVs (richness)",
       color = "severity") +
  scale_x_continuous(expand = c(0, 0), labels = scales::comma) +
  scale_y_continuous(expand = c(0, 0))

# Save plot
ggsave("Fungi_Rarefcurves.png", Fungi_curves, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)

```

## Diagnostic number reads (after QC and filtering ampliseq)
```{r}
# Get out controls
ps.ITS.Pythium <- ps.ITS.path %>% ps_filter(experiment != "PM", severity != "seedling", run != "1") 
sample_data(ps.ITS.Pythium)


#Get reads count
sdt.ITS = data.table(as(sample_data(ps.ITS.Pythium), "data.frame"),
                 TotalReads = sample_sums(ps.ITS.Pythium), keep.rownames = TRUE)
setnames(sdt.ITS, "rn", "SampleID")
sdt.ITS$TotalReads <- as.numeric(sdt.ITS$TotalReads)

sdt.ITS %>% 
  arrange(TotalReads) 

```

```{r}
TotalReads_experiment <- sdt.ITS %>% 
                        group_by(experiment) %>%
                            summarize(Total = sum(TotalReads))
```

```{r}
MeanReads_experiment <- sdt.ITS %>% 
                        group_by(experiment) %>%
                            summarize(Mean = mean(TotalReads))
```

### EXTRACT Phylum=NAs for further analysis
```{r}

### Save and Export fasta file with sequences from Phylum=NAs
Pythium.ITS_NAs <- ps.ITS.Pythium %>% subset_taxa(Phylum == "")

taxa.ITS.Pythium_NAs <- tax_table(Pythium.ITS_NAs)

df_taxa.ITS.PythiumNAs <- as.data.frame(taxa.ITS.Pythium_NAs)

#write.table(df_taxa.ITS.PythiumNAs, file="Pythium.ITS_NAs.txt") 

# Extract ASV names and sequences 
seqs.ITS.Py <- Biostrings::DNAStringSet(as.vector(df_taxa.ITS.PythiumNAs$sequence)) 
names(seqs.ITS.Py) <- row.names(df_taxa.ITS.PythiumNAs) 

#Write table
Biostrings::writeXStringSet(seqs.ITS.Py, filepath = "ASV_sequences_Pythium_PhylumNAs.fasta", width = 1000)


```

## CSS normalization
```{r, warning=FALSE, message=FALSE, echo=FALSE}

#Convert the phyloseq object to a metagenomeseq object:
mgs_cssF <- phyloseq_to_metagenomeSeq(ps.ITS.Pythium)

# Perform the Cumulative Sum Scaling:
mgs_cssF <- cumNorm(mgs_cssF)

# Extract the counts and add them to a separate phyloseq object:
css_countsF <- MRcounts(mgs_cssF, norm = TRUE)
ps_cssF <- ps.ITS.Pythium
otu_table(ps_cssF) <- otu_table(t(css_countsF), taxa_are_rows = FALSE)

#Now lets compare the original data to the CSS normalized data:
otu_table(ps.ITS.Pythium)[1:5, 1:5]
head(sample_sums(ps.ITS.Pythium))

otu_table(ps_cssF)[1:5, 1:5]
head(sample_sums(ps_cssF))

```
## Alpha diversity
```{r, warning=FALSE, message=FALSE, echo=FALSE}

# Alpha diversity - Evenness
Evenness.ITS.Pythium <- microbiome::evenness(ps.ITS.Pythium, index = "all")

# Create data frame with evenness measurements
df.ITS.Pythium.Ev <- Evenness.ITS.Pythium[,c('pielou', 'simpson')]

# All alpha diversity indexes estimation
richness.ITS.Pythium <- estimate_richness(ps.ITS.Pythium, measures = c("Observed", "Shannon", "InvSimpson", "Chao1"))
                                  
#Create a data frame with All alpha diversity and metadata
adiv.ITS.Pythium <- cbind(sample_data(ps.ITS.Pythium), richness.ITS.Pythium, df.ITS.Pythium.Ev)

#write.table(adiv.ITS.Pythium, file="adiv.all.ITS.PythiumResults.txt") 

#adiv.ITS.Pythium <- read.table("adiv.all.ITS.PythiumResults.txt")
```

```{r}
### Means and standard deviation diversity indexes by disease severity
mean_sd_adiv.ITS <- adiv.ITS.Pythium %>%
  group_by(severity, run) %>%
  summarise_at(vars(c(Observed, Shannon, InvSimpson, Chao1, pielou)), list(name = mean, sd))

#write.table(mean_sd_adiv.ITS, file="mean_sd_adiv.ITSPythium.txt")
```

```{r}
################### Statistics - alpha diversity ################################
### Diagnostic models

####### Observed (richness)
model.Observed <- lm(Observed ~ severity * run, data = adiv.ITS.Pythium)
Anova(model.Observed) #significant effect all and interaction (p-val=0.016917 *)

###Analysis by run
#Run 1
adiv.ITS.PythiumR1 <- adiv.ITS.Pythium %>% filter(run == "2")

model.ObservedR1 <- lmer(Observed ~ severity + (1|pond), data = adiv.ITS.PythiumR1)
Anova(model.ObservedR1) #<significant effect of severity p-val=0.0000004116 ***

#Residuals plots
par(mfrow = c(2, 3))
plot(model.ObservedR1, which=1:6) 
res <- resid(model.ObservedR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal  p-value = 0.5646

# Pairwise comparisons
mObs <- emmeans(model.ObservedR1, "severity")
pairs(mObs) #significant differences between control-high and high-low

multcomp::cld(object = mObs, Letters = letters)

#Other way of posthoc test
emmeans(model.ObservedR1, list(pairwise ~ severity), adjust = "tukey")

##NOTE other options for lmer models:
#You could use emmeans::emmeans() or lmerTest::difflsmeans(), or multcomp::glht()


#####Run 2
adiv.ITS.PythiumR2 <- adiv.ITS.Pythium %>% filter(run == "3")

model.ObservedR2 <- lmer(Observed ~ severity + (1|pond), data = adiv.ITS.PythiumR2)
Anova(model.ObservedR2) #significant effect of severity p-val=0.01869 *

#Residuals plots
par(mfrow = c(2, 3))
plot(model.ObservedR2, which=1:6) 
res <- resid(model.ObservedR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal  p-value = 0.2593

# Pairwise comparisons
mObs <- emmeans(model.ObservedR2, "severity")
pairs(mObs) #NO significant differences between any?

multcomp::cld(object = mObs, Letters = letters)

#Other way of posthoc test
emmeans(model.ObservedR2, list(pairwise ~ severity), adjust = "tukey")

# poshoc
compare_means(Observed ~ severity, data = adiv.ITS.PythiumR2, method = "anova")  
#sig difference p-val=0.017

fm1 <- aov(Observed ~ severity, data = adiv.ITS.PythiumR2)
summary(fm1)
thsd <- TukeyHSD(fm1, "severity", ordered = TRUE)
tidy(thsd)
#sig difference high-control


```


```{r}
######### Simpson index
model.Simpson <- lm(InvSimpson ~ severity * run, data = adiv.ITS.Pythium) 
Anova(model.Simpson)  # significant effect of severity only (p-val=0.0007035 ***)

###Run1 
model.SimpsonR1 <- lmer(InvSimpson ~ severity + (1|pond), data = adiv.ITS.PythiumR1) 
Anova(model.SimpsonR1) # significant effect of severity p-val=0.008711 **

#Residuals plots
plot(model.SimpsonR1, which=1:6) 
res <- resid(model.SimpsonR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.1484

# Pairwise comparisons
mSS <- emmeans(model.SimpsonR1, "severity")
pairs(mSS) #Almost sig diff high-low

multcomp::cld(object = mSS, Letters = letters)

compare_means(InvSimpson ~ severity, data = adiv.ITS.PythiumR1, method = "t.test")
#sig difference control-high

###Run 2
model.SimpsonR2 <- lmer(InvSimpson ~ severity + (1|pond), data = adiv.ITS.PythiumR2) 
Anova(model.SimpsonR2) #significant effect of severity p-val=0.03393 *

#Residuals plots
plot(model.SimpsonR2, which=1:6) 
res <- resid(model.SimpsonR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #NOT Normal p-value = 0.001542


# poshoc
compare_means(InvSimpson ~ severity, data = adiv.ITS.PythiumR2, method = "kruskal.test")  
#sig difference 

pairwise.wilcox.test(adiv.ITS.PythiumR2$InvSimpson, adiv.ITS.PythiumR2$severity, p.adjust.method = "bonferroni")
#sig difference high-control

```


```{r}
####### Shannon index
model.Shannon <- lm(Shannon ~ severity * run, data = adiv.ITS.Pythium)
Anova(model.Shannon) #significant effect of severity only (p-val=0.0002061 ***


##Run 1
model.ShannonR1 <- lmer(Shannon ~ severity + (1|pond), data = adiv.ITS.PythiumR1)
Anova(model.ShannonR1) #significant effect of severity p-val= 0.01401 *


#Residuals plots
plot(model.ShannonR1, which=1:6) 
res <- resid(model.ShannonR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.06013


# Pairwise comparisons
mShaS <- emmeans(model.ShannonR1, "severity")
pairs(mShaS) #No Sig dif ??

emmeans(model.ShannonR1, list(pairwise ~ severity))

# poshoc
compare_means(Shannon ~ severity, data = adiv.ITS.PythiumR1, method = "anova")  
#sig difference p-val=0.0091

fm1 <- aov(Shannon ~ severity, data = adiv.ITS.PythiumR1)
summary(fm1)
thsd <- TukeyHSD(fm1, "severity", ordered = TRUE)
tidy(thsd)
#sig difference high-control


##Run 2
model.ShannonR2 <- lmer(Shannon ~ severity + (1|pond), data = adiv.ITS.PythiumR2)
Anova(model.ShannonR2) #significant effect of severity p-val= 0.01752 *


#Residuals plots
plot(model.ShannonR2, which=1:6) 
res <- resid(model.ShannonR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.05839


# Pairwise comparisons
mShaS <- emmeans(model.ShannonR2, "severity")
pairs(mShaS) #No Sig dif ??

emmeans(model.ShannonR2, list(pairwise ~ severity))

# poshoc
compare_means(Shannon ~ severity, data = adiv.ITS.PythiumR2, method = "anova")  
#sig difference p-val=0.02

fm1 <- aov(Shannon ~ severity, data = adiv.ITS.PythiumR2)
summary(fm1)
thsd <- TukeyHSD(fm1, "severity", ordered = TRUE)
tidy(thsd)
#sig difference high-control

```


```{r}
##### Evenness (Pielou index)
model.Ev <- lm(pielou ~ severity * run, data = adiv.ITS.Pythium)
Anova(model.Ev) # significant effect of severity only (p-val=0.0003762 ***)

##Run 1
model.EvR1 <- lmer(pielou ~ severity + (1|pond), data = adiv.ITS.PythiumR1)
Anova(model.EvR1) # significant effect of severity (p-value=0.01014 *)

#Residuals plots
plot(model.EvR1, which=1:6) 
res <- resid(model.EvR1)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)  ###A Bimodal distribution is observed correlates with each run (potential differences only in one run)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.1297

# Pairwise comparisons


# poshoc
compare_means(pielou ~ severity, data = adiv.ITS.PythiumR1, method = "anova")  
#sig diff p-val=0.012

fm1 <- aov(pielou ~ severity*pond, data = adiv.ITS.PythiumR1)
summary(fm1)
thsd <- TukeyHSD(fm1, "severity", ordered = TRUE)
tidy(thsd)
#sig difference high-control


##Run 2
model.EvR2 <- lmer(pielou ~ severity + (1|pond), data = adiv.ITS.PythiumR2)
Anova(model.EvR2) # significant effect of severity (p-value=0.02872 *)

#Residuals plots
plot(model.EvR2, which=1:6) 
res <- resid(model.EvR2)

par(mfrow = c(1, 2)) # combine plots
hist(res, breaks = 40)  ###A Bimodal distribution is observed correlates with each run (potential differences only in one run)

# QQ-plot
qqPlot(res,
       id = TRUE
      )

# Normality test (normal if p>0.05)
shapiro.test(res) #Normal p-value = 0.1285

# Pairwise comparisons
mES <- emmeans(model.EvR2, "severity")
pairs(mES)  #No Sig dif because interactions
multcomp::cld(object = mES, Letters = letters)

# poshoc
compare_means(pielou ~ severity, data = adiv.ITS.PythiumR2, method = "anova")  
#sig diff p-val=0.029

fm1 <- aov(pielou ~ severity, data = adiv.ITS.PythiumR2)
summary(fm1)
thsd <- TukeyHSD(fm1, "severity", ordered = TRUE)
tidy(thsd)

```


#### Boxplot
```{r}
#Boxplot observed alpha diversity 
adiv.ITS.Pythium$severity <- ordered(adiv.ITS.Pythium$severity, levels = c("low", "control", "high"))

# Prepare a vector of colors with specific color for sample type
library(RColorBrewer)
myColors <- c("palegreen1", "#1F78B4", "hotpink1")

#Observed plot 
observed.ITS.Pythium.plot <- ggplot(adiv.ITS.Pythium, aes(x = severity, y = Observed, fill=severity)) +
  geom_boxplot() + 
  ylab('Fungi Richness (Observed)') +
  xlab('Disease Severity') +
  scale_fill_manual(name = "Severity", values = myColors) +
  labs(fill = 'severity') +
  theme(axis.text.x = element_text(angle = 45, hjust=1))

#Plot Shannon
Shannon.ITS.Pythium.plot <- ggplot(adiv.ITS.Pythium, aes(x = severity, y = Shannon, fill=severity)) + geom_boxplot(outlier.shape = NA) +
  ylab('Shannon index') +
  xlab('Disease Severity') +
  scale_fill_manual(name = "severity", values = myColors) + 
    theme_few() +
    theme(legend.position = "none")

# Save plot
ggsave("Pythium_ShannonFungi.png", Shannon.ITS.Pythium.plot, 
  width = 11,
  height = 12,
  units = "cm",
  dpi = 600)

#Plots Evenness 
PielouEv.ITS.Pythium.plot <- ggplot(adiv.ITS.Pythium, aes(x = severity, y = pielou, fill=severity)) +
  geom_boxplot() + 
  ylab('Bacteria Evenness (Pielou)') +
  xlab('Disease Severity') +
  scale_fill_manual(name = "severity", values = myColors) +
  labs(fill = 'severity') 

# Merging plots
library(patchwork)

alphaDivITS.Pythium <- observed.ITS.Pythium.plot + PielouEv.ITS.Pythium.plot +
  #plot_annotation(tag_levels = "A") +
    plot_layout(guides = 'collect') & 
    theme (axis.title = element_text (size = 10)) &
      theme(plot.tag = element_text(size = 12)) &
          theme(axis.text.y = element_text (size = 8)) &
            scale_x_discrete(name ="Disease severity") 


```

## Beta Diversity
```{r, warning=FALSE, message=FALSE, echo=FALSE}
#Get relative abundance after normalization
#ps_cssF_RA <- transform_sample_counts(ps_cssF, function(x) x/sum(x))

ps.ITS.Pythium <- readRDS('ps.ITS.PythiumNC.rds')

ps.ITS.Pythium_RA <- transform_sample_counts(ps.ITS.Pythium, function(x) x/sum(x))

#Distance
sampledfF.Py <- data.frame(sample_data(ps.ITS.Pythium_RA))
dist.matF.Py <- phyloseq::distance(ps.ITS.Pythium_RA, method = "bray")

set.seed(100)
perm <- how(nperm = 999)
#setBlocks(perm) <- with(sampledf, run)

pmv.PyF <- adonis2(dist.matF.Py ~ severity + run, data = sampledfF.Py, permutations = 999, by = "margin")
print(pmv.PyF) #No significant interaction p-val=0.545

## Permanova poshoc NOT APPLICABLE SINCE PERMANOVA NOT SIGNIFICANT
set.seed(100)
pairwise.perm.manova(dist.matF.Py, sampledfF.Py$severity, nperm = 999, progress = TRUE, p.method = "fdr", F = T, R2 = T) #high-control=0.003, #low-control=0.019 #high-low=0.132

set.seed(100)
pairwise.perm.manova(dist.matF.Py, sampledfF.Py$run, nperm = 999, progress = TRUE, p.method = "fdr", F = T, R2 = T) #p-val=0.015
```

### PCoA plot
```{r}
#Color palette
myColors2 <- c("#1F78B4", "hotpink1", "palegreen1")

#Calculate ordinations
ordBCF.Py = ordinate(ps_cssF_RA, method = "PCoA", distance = dist.matF.Py)

## Plot
p.Ord.BCF.Py <- plot_ordination(ps_cssF_RA, ordBCF.Py, color = "severity") 

Bdiv.ploF.Py <- p.Ord.BCF.Py + 
  geom_point(mapping = aes(shape = factor(run)),size=5, alpha=0.85) +
    stat_ellipse() +
      geom_vline(xintercept = c(0), color = "grey70", linetype = 2) +
        geom_hline(yintercept = c(0), color = "grey70", linetype = 2) +
          theme_bw() + 
            theme(panel.background = element_blank(), 
              panel.grid = element_blank(), 
              legend.title=element_text(size=14),
              legend.text=element_text(size=12)) + 
  scale_color_manual(name = "Disease severity", values = myColors2, breaks =c("control", "high", "low"), labels = c("Control", "High", "Low")) +
  scale_shape_discrete(labels=c("Run 1","Run 2"),name="Experiment")
                                    
# Save plot
ggsave("PCoA_BrayF_Pyhtium.png", Bdiv.ploF.Py, 
  width = 15,
  height = 12,
  units = "cm",
  dpi = 600)


```

## Beta-dispersion test
```{r}
# calculates the beta-dispersion for each group, when comparing 2 or more
#For severity
disp.testSev <- betadisper(dist.matF.Py, group = sampledfF.Py$severity, bias.adjust = TRUE)

#For inoculation
disp.testInc <- betadisper(dist.matF.Py, group = sampledfF.Py$inoculation, bias.adjust = TRUE)

#For run
disp.testRun <- betadisper(dist.matF.Py, group = sampledfF.Py$run, bias.adjust = TRUE)

# tests if centroid distances are significantly different from each other
disp.anovaSev <- anova(disp.testSev) #Result= centroids sig different 0.008123 ** 

disp.anovaInc <- anova(disp.testInc) #Result= centroids sig different p-value=0.007701 **

disp.anovaRun <- anova(disp.testRun) #Result= centroids NO sig different p-value=0.05957 .
```

```{r}
# test significance between each group
dispSev.TukeyHSD <- TukeyHSD(disp.testSev) #sig diff high-control p-adj= 0.0144541, low-control p-adj=0.0175967
dispInc.TukeyHSD <- TukeyHSD(disp.testInc) #sig diff p-adj=0.0006948

dispRun.TukeyHSD <- TukeyHSD(disp.testRun) #NO sig diff runs p-adj=0.972428

```

```{r}
# plot showing the dispersion for each group
plot(disp.testInc, hull = FALSE, ellipse = TRUE)
```

```{r}
#test for homogeneity of multivariate dispersions 
set.seed(100)
permutest(disp.testInc, pairwise = TRUE, permutations = 999) #0.006 *
```


## Differential abundance analysis
```{r, warning=FALSE, message=FALSE, echo=FALSE}
# Aglomerate taxa
#ps.ITS.Pythium <- tax_glom(ps.ITS.Pythium, "Genus")
#df <- psmelt(ps.ITS.Pythium)

#Fix taxa names
#we want to add the genus, and if not available, the last taxonomic information available.
ps.ITS.Pythium <- tax_fix(ps.ITS.Pythium)

# Converting phyloseq object to DESeq2 format 
dds.ITS.Pythium.df <- phyloseq_to_deseq2(ps.ITS.Pythium, ~severity + run)

# Identifying the groups in our experiment
dds.ITS.Pythium.df$Group <- factor(paste0(dds.ITS.Pythium.df$severity, dds.ITS.Pythium.df$run))  

# Setting our experimental design in the DESeq2 object
design(dds.ITS.Pythium.df) <- formula(~Group) 

# Generating our analysis
dds.analysis.ITS <- DESeq(dds.ITS.Pythium.df, quiet = TRUE) 

results(dds.analysis.ITS)

# To check contrasting groups
levels(dds.analysis.ITS$Group)

### Select the groups to contrast 
# Contrast Control vs. High in Run 2
res.CvsH.run2 <- results(dds.analysis.ITS, contrast = c("Group", "control2", "high2")) 
# Contrast Control vs. Low in Run 2
res.CvsL.run2 <- results(dds.analysis.ITS, contrast = c("Group", "control2", "low2")) 

# Contrast Control vs. High in Run 3
res.CvsH.run3 <- results(dds.analysis.ITS, contrast = c("Group", "control3", "high3")) 
# Contrast Control vs. Low in Run 3
res.CvsL.run3 <- results(dds.analysis.ITS, contrast = c("Group", "control3", "low3"))

```

```{r}
## RESULTS CONTRAST: Contrast Control vs. High in Run 2
# First look at the results 
summary(res.CvsH.run2) 

# Number of taxa with significantly higher abundance in High run2:
sum(res.CvsH.run2$padj < 0.1 & res.CvsH.run2$log2FoldChange > 0, na.rm = TRUE)  #Result 4

# Number of taxa with significantly lower abundance in High run2:
sum(res.CvsH.run2$padj < 0.1 & res.CvsH.run2$log2FoldChange < 0, na.rm = TRUE)  #Result 26

# To look the most significantly different taxa
res.CvsH.run2.order <- res.CvsH.run2[order(res.CvsH.run2$pvalue), ]

# Create a data frame from your DESeq results 
res.CvsH.run2.df <- as.data.frame(res.CvsH.run2)

### Add the taxonomy of the ASVs to the new data frame 
CvsH.run2.taxmelt <- as.data.frame(tax_table(ps.ITS.Pythium))
## now let us apply a the prefix to each one of them
res.CvsH.run2.df$Tax <- paste(rownames(res.CvsH.run2.df), CvsH.run2.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsH.run2.df$association = ifelse(res.CvsH.run2.df$padj <= 0.05 & res.CvsH.run2.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsH.run2.df$association = ifelse(res.CvsH.run2.df$padj <= 0.05 & res.CvsH.run2.df$log2FoldChange < 0, "Low_abundant", res.CvsH.run2.df$association)

# Filter to keep only log2folds > |5| and p-values <= 0.05 
CvsH.run2.res05 <- dplyr::filter(res.CvsH.run2.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Making sure our new table has everything we need 
head(CvsH.run2.res05)

# Generating a dataframe from our table
CvsH.run2.res05.df <- as.data.frame(CvsH.run2.res05)

#Add new variables for categories used while plotting
CvsH.run2.res05.df$Severity <- "High"
CvsH.run2.res05.df$Exp_run <- "Run2"

## Bar plot 
CvsH.run2.plot <- ggplot(data = CvsH.run2.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Fungal Genera in High Disease Severity vs. Control - Run2") +
  theme(plot.title = element_text(hjust = 0.5))

```

```{r}
## RESULTS CONTRAST: Contrast Control vs. Low in Run 2

# First look at the results 
summary(res.CvsL.run2)

# Number of taxa with significantly higher abundance in Low run2:
sum(res.CvsL.run2$padj < 0.1 & res.CvsL.run2$log2FoldChange > 0, na.rm = TRUE)  #Result 0

# Number of taxa with significantly lower abundance in Low run2:
sum(res.CvsL.run2$padj < 0.1 & res.CvsL.run2$log2FoldChange < 0, na.rm = TRUE)  #Result 0

# To look the most significantly different taxa
res.CvsL.run2.order <- res.CvsL.run2[order(res.CvsL.run2$pvalue), ]
head(res.CvsL.run2.order)

# Create a data frame from your DESeq results 
res.CvsL.run2.df <- as.data.frame(res.CvsL.run2)

# Add the taxonomy of the ASVs to the new data frame 
CvsL.run2.taxmelt <- as.data.frame(tax_table(ps.ITS.Pythium))
res.CvsL.run2.df$Tax <- paste(rownames(res.CvsL.run2.df), CvsL.run2.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsL.run2.df$association = ifelse(res.CvsL.run2.df$padj <= 0.05 & res.CvsL.run2.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsL.run2.df$association = ifelse(res.CvsL.run2.df$padj <= 0.05 & res.CvsL.run2.df$log2FoldChange < 0, "Low_abundant", res.CvsL.run2.df$association)


# Filter to keep only log2folds > |5| and pvalues <= 0.05 
CvsL.run2.res05 <- dplyr::filter(res.CvsL.run2.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Making sure our new table has everything we need 
head(CvsL.run2.res05)

# Generating a dataframe from our table
CvsL.run2.res05.df <- as.data.frame(CvsL.run2.res05)

#Add new variables for categories used while plotting
CvsL.run2.res05.df$Severity <- "Low"
CvsL.run2.res05.df$Exp_run <- "Run2"

## Plot 
CvsL.run2.plot <- ggplot(data = CvsL.run2.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Fungal Genera in Low Disease Severity vs. Control - Run2") +
  theme(plot.title = element_text(hjust = 0.5))

```

### Barplot High&Low Run2
```{r}
# Join dataframes with DeSeq results
BindDF.run2F <- bind_rows(CvsH.run2.res05.df, CvsL.run2.res05.df)

##Save final dataframe
write.table(BindDF.run2F, file = "DESeq_Pythium_dataRun2F.tsv", sep = "\t", row.names = TRUE)

## Plot (normal barplot - don't show all ASVs from low and high disease severity)
finalRun2F.plot <- ggplot(data = BindDF.run2F,
       aes(x = log2FoldChange, y = Tax, fill = Severity, colour = association)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  geom_vline(xintercept = c(0), color = "grey0", linetype = 2) +
  labs(title = "Experiment Run 1") +
  scale_fill_manual(name = "Disease severity", values = c("white", "grey")) +
  scale_color_manual(name = "Abundance", values = c("tomato1", "steelblue3"), breaks = c("High_abundant", "Low_abundant"), labels = c("Enriched in inoculated roots", "Depleted in inoculated roots")) + labs(y = "\nASV:taxa") +
   theme_few() +
  theme(plot.title = element_text(size = 12),
  theme(axis.text.x = element_text(angle = 90, size = 13),
        legend.title=element_text(size=13),
        legend.text=element_text(size=12))) 
 
 

# Save plot
ggsave("DiffAbundFungi-PythiumR2.png", finalRun2F.plot, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)
  
```

```{r}
## RESULTS CONTRAST: Contrast Control vs. High in Run 3

# First look at the results 
summary(res.CvsH.run3)

# Create a data frame from your DESeq results 
res.CvsH.run3.df <- as.data.frame(res.CvsH.run3)

# Add the taxonomy of the ASVs to the new data frame 
CvsH.run3.taxmelt <- as.data.frame(tax_table(ps.ITS.Pythium))
res.CvsH.run3.df$Tax <- paste(rownames(res.CvsH.run3.df), CvsH.run3.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsH.run3.df$association = ifelse(res.CvsH.run3.df$padj <= 0.05 & res.CvsH.run3.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsH.run3.df$association = ifelse(res.CvsH.run3.df$padj <= 0.05 & res.CvsH.run3.df$log2FoldChange < 0, "Low_abundant", res.CvsH.run3.df$association)


# Filter to keep only log2folds > |5| and pvalues <= 0.05 
CvsH.run3.res05 <- dplyr::filter(res.CvsH.run3.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Generating a dataframe from our table
CvsH.run3.res05.df <- as.data.frame(CvsH.run3.res05)

#Add new variables for categories used while plotting
CvsH.run3.res05.df$Severity <- "High"
CvsH.run3.res05.df$Exp_run <- "Run3"

## Plot 
CvsH.run3.plot <- ggplot(data = CvsH.run3.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Fungal Genera in High Disease Severity vs. Control - Run3") +
  theme(plot.title = element_text(hjust = 0.5))

```

```{r}
## RESULTS CONTRAST: Contrast Control vs. Low in Run 3

# First look at the results 
summary(res.CvsL.run3)

# Create a data frame from your DESeq results 
res.CvsL.run3.df <- as.data.frame(res.CvsL.run3)

# Add the taxonomy of the ASVs to the new data frame 
CvsL.run3.taxmelt <- as.data.frame(tax_table(ps.ITS.Pythium))
res.CvsL.run3.df$Tax <- paste(rownames(res.CvsL.run3.df), CvsL.run3.taxmelt$Genus, sep= ":")

# Add an association variable for the volcano and lollipop plots
res.CvsL.run3.df$association = ifelse(res.CvsL.run3.df$padj <= 0.05 & res.CvsL.run3.df$log2FoldChange > 0, "High_abundant", "Not significant")
res.CvsL.run3.df$association = ifelse(res.CvsL.run3.df$padj <= 0.05 & res.CvsL.run3.df$log2FoldChange < 0, "Low_abundant", res.CvsL.run3.df$association)


# Filter to keep only log2folds > |5| and p-values <= 0.05 
CvsL.run3.res05 <- dplyr::filter(res.CvsL.run3.df, padj<=0.05, log2FoldChange<=-5 | log2FoldChange>=5) 

# Generating a dataframe from our table
CvsL.run3.res05.df <- as.data.frame(CvsL.run3.res05)

#Add new variables for categories used while plotting
CvsL.run3.res05.df$Severity <- "Low"
CvsL.run3.res05.df$Exp_run <- "Run3"

## Plot 
CvsL.run3.plot <- ggplot(data = CvsL.run3.res05.df,
       aes(x = Tax, y = log2FoldChange, fill = log2FoldChange)) +
  geom_bar(stat="identity") +
  coord_flip() +
  labs(title = "Differentially abundant Fungal Genera in Low Disease Severity vs. Control - Run3") +
  theme(plot.title = element_text(hjust = 0.5))

```

### Barplot High&Low Run3
```{r}
# Join dataframes with DeSeq results
BindDF.run3F <- bind_rows(CvsH.run3.res05.df, CvsL.run3.res05.df)

##Save final dataframe
write.table(BindDF.run3F, file = "DESeq_Pythium_dataRun3F.tsv", sep = "\t", row.names = TRUE)

## Plot (normal barplot - don't show all ASVs from low and high disease severity)
finalRun3F.plot <- ggplot(data = BindDF.run3F,
       aes(x = log2FoldChange, y = Tax, fill = Severity, colour = association)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  geom_vline(xintercept = c(0), color = "grey0", linetype = 2) +
  labs(title = "Experiment Run 2") +
  scale_fill_manual(name = "Disease severity", values = c("white", "grey")) +
  scale_color_manual(name = "Abundance", values = c("tomato1", "steelblue3"), breaks = c("High_abundant", "Low_abundant"), labels = c("Enriched in inoculated roots", "Depleted in inoculated roots")) + labs(y = "\nASV:taxa") +
   theme_few() +
  theme(plot.title = element_text(size = 12),
  theme(axis.text.x = element_text(angle = 90, size = 13),
        legend.title=element_text(size=13),
        legend.text=element_text(size=12))) 
 
 

# Save plot
ggsave("DiffAbundFungi-PythiumR3.png", finalRun3F.plot, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)
  
```
### All Barplots  (did not work!!)
```{r}
# Join dataframes with DeSeq results
BindDF.run23F <- bind_rows(CvsH.run2.res05.df, CvsL.run2.res05.df, CvsH.run3.res05.df, CvsL.run3.res05.df)

# Tidy dataframe
data.run23F <- BindDF.run23F %>%
  dplyr::group_by(Tax, association, Severity, Exp_run) %>% 
  dplyr::summarize(log2FoldChange = mean(log2FoldChange))

data.run23F <- aggregate(log2FoldChange ~ ., BindDF.run23F, sum)

##Save final dataframe
write.table(data.run23F, file = "DESeq_Pythium_dataRun23F.tsv", sep = "\t", row.names = TRUE)

## Plot (normal baplot - don't show all ASVs from low and high disease severity)
finalRun23F.plot <- ggplot(data = data.run23F,
       aes(x = log2FoldChange, y = Tax, fill = Severity)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  coord_flip() +
  facet_grid(~Exp_run) +
  geom_vline(xintercept = c(0), color = "grey70", linetype = 2) +
  labs(title = "Differentially abundant Fungi") +
  scale_fill_manual(name = "Disease severity", values = c("tomato1", "lightsalmon")) + labs(y = "\nGenera") + 
  theme(strip.background = element_rect(color="black", fill="gray", size=1, linetype="solid")) +
  theme_bw() +
  theme(plot.title = element_text(size = 12)) +
  theme(axis.text.x = element_text(angle = 90, size = 7))
 

# Save plot
ggsave("DiffAbundFungi-PythiumAll.png", finalRun23F.plot, 
  width = 16,
  height = 14,
  units = "cm",
  dpi = 600)
  
```
# Pythium sequences analysis
```{r}
#Importing DADA table (output from ASV inference - ampliseq)
ps.Seqs <- readRDS('DADA2_table_ITS.rds')

########### Assign taxonomy #################

# FASTA file with an Oomycetes database (created by Antonino using ref seqs from NCBI)
# sequences will be compared to it
tax_key <- "dada2_Oomycetes_BOLD_20242207.fasta" 

#Current UNITE general FASTA release for eukaryotes
tax_key2 <- "sh_general_release_dynamic_all_19.02.2025.fasta"   

# Assign taxonomy to the ASVs

taxa_Oomy <- assignTaxonomy(ps.Seqs, tax_key, multi = TRUE, verbose = TRUE)
saveRDS(taxa_Oomy, "Oomycetes_taxAssign.rds")

#taxa_Oomy <- readRDS('Oomycetes_taxAssign.rds')

taxa_Unite <- assignTaxonomy(ps.Seqs, tax_key2, multi = TRUE, verbose = TRUE)
saveRDS(taxa_Unite, "UNITE_taxAssign.rds")

View(tax2)

#check whether the sequences in our taxonomy table and sequence table match
if(!identical(getSequences(taxa_Oomy), getSequences(ps.Seqs))) stop("Taxonomy mismatch.")

# How many ASVs are in each genus?
table(taxa_Oomy[,"Genus"], useNA = "ifany")

##RESULT= Albugo(18), Hyaloperonospora (7), Peronospora (35), Phytophthora (16), Plasmopara (100), Pythium (51) NAs(1601), Pilasporangium (1)

# Fix the sample IDs in the sequence table
rownames(ps.Seqs) <- sub("_1.filt.fastq.gz", "", rownames(ps.Seqs))

# File with sample metadata (data about your samples, treatments, blocking, etc):
metadata_file <- "ITSmetadata_Pathogens.tsv"

# Read and prepare the metadata
metadata_df <- read.table(file = metadata_file, sep = "\t", header = TRUE)

colnames(metadata_df)[1] <- "SampleID"
rownames(metadata_df) <- metadata_df$SampleID

head(metadata_df)

# Create the phyloseq object
ps.Oomycetes.final <- phyloseq(otu_table(ps.Seqs, taxa_are_rows = FALSE),
               sample_data(metadata_df),
               tax_table(taxa_Oomy))

# Saves the phyloseq object as an .rds file (which can be imported directly by phyloseq):
#saveRDS(ps.Oomycetes.final, "ps.Oomycetes_all.rds")

# Load phyloseq object
ps.Oomycetes.final <- readRDS('ps.Oomycetes_all.rds')
```


```{r}
########## Plot Taxonomic composition ##############
# Sub-setting by Experiment
ps.Oomy_Pythium <- ps.Oomycetes.final %>% ps_filter(experiment == "Pythium", run != "1", severity !="seedling")
sample_data(ps.Oomy_Pythium)

# get Relative abundance in %
ps.Oomy_Pythium_RA <- transform_sample_counts(ps.Oomy_Pythium, function(x) x/sum(x))

# Agglomerate taxa by genus level
ps.Oomy.G <- tax_glom(ps.Oomy_Pythium_RA, taxrank = 'Genus', NArm = FALSE)

tax_table(ps.Oomy.G)

# Create dataframe from phyloseq object
data_ps <- psmelt(ps.Oomy.G) 

#Filter and get abundance
filt.gen <- data_ps %>% group_by(Genus) %>% summarise(mean = mean(Abundance))

dat <- subset(data_ps, !(OTU %in% filt.gen$Genus))
dat <- dat %>% group_by(run, severity, Genus) %>% summarise(cs = mean(Abundance)) %>% mutate(cs = cs/sum(cs))
```


```{r}
# Renove NAs (that might be Fungi sequences)
# Agglomerate taxa by genus level
ps.Oomy.G2 <- tax_glom(ps.Oomy_Pythium_RA, taxrank = 'Genus', NArm = TRUE)

# Create dataframe from phyloseq object
data_ps2 <- psmelt(ps.Oomy.G2) 

#Filter and get abundance
filt.gen2 <- data_ps2 %>% group_by(Genus) %>% summarise(mean = mean(Abundance))

dat2 <- subset(data_ps2, !(OTU %in% filt.gen$Genus))
dat2 <- dat2 %>% group_by(run, severity, Genus) %>% summarise(cs = mean(Abundance)) %>% mutate(cs = cs/sum(cs))
```


```{r}
#Bar plot by Phylum level
# Set color palette to accommodate the number of phylum
colourCount = length(unique(data_ps2$Genus))
set.seed(2020)
taxaColors <- distinctColorPalette(colourCount)

# Plot
plot.Genus.Oomycetes <- ggplot(data=data_ps2, aes(x=severity, y=Abundance, fill = fct_reorder(Genus, Abundance))) +
  geom_bar(aes(), stat="identity") +
  scale_fill_manual(values=taxaColors) +
  labs(x = "Disease severity", y = "Relative Abundance\n", fill = "Treatment") +
  guides(fill=guide_legend(title="Genus")) + 
  theme(legend.position="right")

# Save plot
ggsave("Taxa.Genus.Oomycetes.png", plot.Genus.Oomycetes, 
  width = 11,
  height = 12,
  units = "cm",
  dpi = 600)

```

```{r}
### Save and Export fasta file with sequences from Genus=Pythium (to BLAST with Sanger sequences of PA1)

Pythium.seqs <- ps.Oomycetes.final %>% subset_taxa(Genus == "Pythium")

taxa.Pythium.seqs <- tax_table(Pythium.seqs)

df_taxa.Pythium.seqs <- as.data.frame(taxa.Pythium.seqs)

names(seqs.Py) <- row.names(df_taxa.Pythium.seqs) 


```

```{r}

# Boxplot of Abundance by disease severity and variety
Abundance_Pythium <- ggplot(data_ps2, aes(x = severity, y = Abundance, fill = severity)) +
  geom_bar(stat = "identity", position = position_dodge()) +
  facet_wrap(~run) +
  theme_minimal() +
  labs(title = "Abundance of Pythium",
       x = "Disease severity",
       y = "Relative Abundance") 

# Save plot
ggsave("Abundance_Pythium.png", Abundance_Pythium, 
  width = 11,
  height = 12,
  units = "cm",
  dpi = 600)
```




